File: PluginExamples/Server/CSharp/BareBones/YourPlugin.cs
using OTA.Plugin;
using OTA.Command;
using OTA.Logging;

namespace BareBones
{
	[OTAVersion(1, 0)]
    public class YourPlugin : BasePlugin
    {
        public YourPlugin()
        {
            this.Version = "1";
            this.Author = "TDSM";
            this.Name = "Simple name";
            this.Description = "This plugin does these awesome things!";
        }

        protected override void Initialized(object state)
        {
			ProgramLog.Plugin.Log ("Your plugin is initialising");
        }

        [Hook(HookOrder.NORMAL)]
        void MyFunctionNameThatDoesntMatter(ref HookContext ctx, ref HookArgs.NpcKilled args)
        {
            //Your implementation
        }
    }
}

File: PluginExamples/Server/VB/BareBones/YourPlugin.vb
Imports OTA.Command
Imports OTA.Plugin
Imports OTA.Logging

<OTAVersion(1, 0)> _
Public Class YourPlugin : Inherits BasePlugin
    Public Sub New()
        Me.Version = "1"
        Me.Author = "TDSM"
        Me.Name = "Simple name"
        Me.Description = "This plugin does these awesome things!"
    End Sub

    Protected Overrides Sub Initialized(state As Object)
    	ProgramLog.Debug.Log ("Your plugin is initialising", False)
    End Sub

    <Hook(HookOrder.NORMAL)> _
    Sub MyFunctionNameThatDoesntMatter(ByRef ctx As HookContext, ByRef args As HookArgs.PlayerEnteredGame)
        'Your implementation
    End Sub

End Class
